### ISSUE TITLE
NVIDIA — Senior Perception Software Engineer

### ISSUE THREAD

**Issue by louisinayinde** [louisinayinde/jobs#486]
**Entreprise** : NVIDIA  
**Poste** : Senior Perception Software Engineer  
**Localisation** : Canada; United States · télétravail  
**Score** : 11 — default · stack +11 (C++, Python)  
**Tech** : C++, Python  
**Lien ATS** : [himalayas.app](<https://himalayas.app/companies/nvidia/jobs/senior-perception-software-engineer>)  
**Publiée le** : 2026-09-20  
**Source** : himalayas  

---

**Description**

NVIDIA has been transforming computer graphics, PC gaming, and accelerated computing for more than 25 years. It’s a unique legacy of innovation that’s fueled by great technology—and amazing people. Today, we’re tapping into the unlimited potential of AI to define the next era of computing. An era in which our GPU acts as the brains of computers, robots, and self-driving cars that can understand the world. Doing what’s never been done before takes vision, innovation, and the world’s best talent. As an NVIDIAN, you’ll be immersed in a diverse, supportive environment where everyone is inspired to do their best work. Come join the team and see how you can make a lasting impact on the world.

Intelligent machines powered by Artificial Intelligence computers that can learn, reason and interact with people are no longer science fiction. GPU Deep Learning has provided the foundation for machines to learn, perceive, reason and solve problems. Now, NVIDIA’s GPU runs Deep Learning algorithms, simulating human intelligence, and acts as the brain of computers, robots and self-driving cars that can perceive and understand the world.

We are now looking for a passionate Software Engineer to design, develop, and productize NVIDIA's autonomous driving solutions. As a member of our perception team, you will build world-class perception solutions. You will design and implement perception SW component and be challenged to improve robustness, efficiency, reusability of the software component working closely with the world class perception engineers and researchers.

What you'llbe doing:

\-

Execute full software development life cycle (SDLC).

\-

Write well-designed, testable code.

\-

Integrate perception (but not limited) software components into a fully functional software system.

\-

Develop software verification plans and quality assurance procedures.

\-

Document and maintain software functionality.

\-

Solve, debug and upgrade existing systems.

\-

Stay in sync with project plans and industry standards

\-

Deploy state-of-the art perception DNN models to embedded platforms.

\-

Analyze and improve the system using both real and synthetic benchmark data.

What we need to see:

\-

BS, MS or PhD in Computer Science or related fields (or equivalent experience).

\-

Proven work experience (12+ years) on embedded platforms for real-time applications as a Software Engineer or Software Developer

\-

Strong programming skills in C++ and Python.

\-

Experience with test-driven development.

\-

Experience with disciplined, design based approach to software development.

\-

Proficiency in software engineering tools.

\-

Ability to document requirements and specifications.

\-

Outstanding communication and collaboration skills as we work as a tightly-knit team, always discussing and learning from each other.

Ways to stand out from the crowd:

\-

Experience with CUDA / SIMD

\-

Hands on background in deploying DNN models to embedded platforms for real-time applications.

\-

Hands-on experience in using simulated / synthetic data to improve accuracy of machine learning models.

\-

Background with deep learning framework (e.g., TensorFlow, PyTorch, etc).

\-

Experience with integration and testing on real hardware / robotic platforms

NVIDIA is widely considered to be one of the technology world’s most desirable employers. We have some of the most forward-thinking and hardworking people in the world working for us. If you're creative and autonomous, we want to hear from you!

Your base salary will be determined based on your location, experience, and the pay of employees in similar positions. The base salary range is 224,000 USD - 356,500 USD.

You will also be eligible for equity and benefits.

Applications for this job will be accepted at least until September 22, 2026.

This posting is for an existing vacancy.

NVIDIA uses AI tools in its recruiting processes.

NVIDIA is committed to fostering an inclusive work environment and proud to be an equal opportunity employer. As we highly value diversity in our current and future employees, we do not discriminate (including in our hiring and promotion practices) on the basis of race, religion, color, national origin, gender, gender expression, sexual orientation, age, marital status, veteran status, disability status or any other characteristic protected by law.

Originally posted on Himalayas
